src/scripts/main.js

```diff
@@ -1,3 +1,323 @@
+          - rowB.cells[colNum].textContent;
+        case 'salary':
+          return toNumber(rowA.cells[colNum].textContent)
+          - toNumber(rowB.cells[colNum].textContent);
+      }
+    };
+  } else {
+    compare = function(rowA, rowB) {
+      switch (type) {
+        case 'string':
+          return rowB.cells[colNum].textContent
+            .localeCompare(rowA.cells[colNum].textContent);
+
+        case 'number':
+          return rowB.cells[colNum].textContent
+          - rowA.cells[colNum].textContent;
+        case 'salary':
+          return toNumber(rowB.cells[colNum].textContent)
+          - toNumber(rowA.cells[colNum].textContent);
+      }
+    };
+  }
+
+  rowsArray.sort(compare);
+  tBody.append(...rowsArray);
+}
+
+function toNumber(string) {
+  const str = string.replace(',', '.').slice(1);
+
+  return str;
+}
+
+const handleListRows = function(ev) {
+  const item = ev.target;
+
+  if (item.tagName !== 'TD') {
+    return;
+  }
+  item.parentElement.classList.toggle('active');
+};
+
+const preventSelection = function(ev) {
+  ev.preventDefault();
+};
+
+table.addEventListener('click', handleListRows);
+table.addEventListener('mousedown', preventSelection);
+
+// adding form
+
+body.insertAdjacentHTML('beforeend', `
+  <form
+    class='new-employee-form'
+    method = "GET"
+    action="/"
+  >
+    <label>
+      Name:
+      <input
+        type="text"
+        name="name"
+        data-qa="name"
+        minlength="4"
+        required
+      />
+    </label>
+    <label>
+      Position:
+      <input
+        type="text"
+        name="position"
+        data-qa="position"
+        minlength="4"
+        required
+      />
+    </label>
+    <label>
+      Office:
+      <select
+        name="office"
+        data-qa="office"
+        required
+      >
+        <option>Tokyo</option>
+        <option>Singapore</option>
+        <option>London</option>
+        <option>New York</option>
+        <option>Edinburgh</option>
+        <option>San Francisco</option>
+      </select>
+    </label>
+    <label>
+      Age:
+      <input
+        type="number"
+        name="age"
+        data-qa="age"
+        min="18"
+        max="90"
+        required
+      />
+    </label>
+    <label>
+      Salary:
+      <input
+        type="number"
+        name="salary"
+        data-qa="salary"
+        required
+      />
+    </label>
+
+    <button type="submit">Save to lable</button>
+      
+  </form>
+`);
+
+const form = document.querySelector('.new-employee-form');
+
+const button = body.querySelector('button');
+
+const inputName = document.querySelector('[data-qa ="name"]');
+const inputPosition = document.querySelector('[data-qa ="position"]');
+const inputOffice = document.querySelector('[data-qa ="office"]');
+const inputAge = document.querySelector('[data-qa ="age"]');
+const inputSalary = document.querySelector('[data-qa ="salary"]');
+
+const onlyWithLabels = [...form].filter(element =>
```

Comment:
The filter() method creates a new array with all elements that pass the test specified in the passed function.
```suggestion
const onlyWithLabels = form.filter(element =>
```

Reply:
![image](https://user-images.githubusercontent.com/95346601/197389283-f8608b74-b86c-4e5a-9e77-d98ed8010ab4.png)

if I write form.filter  and not [...form].filter it gets me error